### PR TITLE
jgrss/store time series

### PR DESCRIPTION
**Pull request by jgrss**
## What is this PR changing?
* Adds support for Dask client in `save()`
* Adds support to write (time x band x height x width)-shaped arrays by scattering the band or time dimension to separate files

### CODE REVIEW

**Comment by mmann1123** [comment]
I am going to try to take a look at this.

File: tests/test_stac.py [hunk 1]
```diff
@@ -28,15 +35,47 @@
...
+        proj_df.buffer(
+            100, cap_style=CAP_STYLE.square, join_style=JOIN_STYLE.mitre
+        )
+        .to_crs('epsg:4326')
+        .to_frame(name='geometry')
+    )
+
+    return df, tuple(proj_df.total_bounds.flatten().tolist())
 
-DF = gpd.GeoDataFrame(geometry=[shape(geojson)], crs=4326)
```
Comment (mmann1123) [inline tests/test_stac.py:33]: should be updated to `shape(search_geojson)` I think.

File: tests/test_stac.py [hunk 2]
```diff
@@ -28,15 +35,47 @@
     ],
 }
 
-EPSG = 8857
```
Comment (mmann1123) [inline tests/test_stac.py:31]: its failing without this line

**Review by mmann1123 — APPROVED** [review]
The client save seems to be working well. Seems like you inadvertently removed some lines that muck up `stac` tests.  I added comments to address

File: src/geowombat/backends/rasterio_.py [hunk 3]
```diff
@@ -164,21 +168,145 @@ class RasterioStore(object):
...
+
+    def _setup(self):
+        if self.scatter is None:
+            if len(self.data.shape) > 3:
+                raise ValueError(
+                    "Only 3-band arrays can be written when scatter=None."
+                )
+
+        if hasattr(self.data, "_FillValue"):
+            nodata = self.data.attrs["_FillValue"]
```
Comment (jgrss) [inline src/geowombat/backends/rasterio_.py:198]: This is no longer a user argument.

File: src/geowombat/backends/rasterio_.py [hunk 4]
```diff
@@ -164,21 +168,145 @@ class RasterioStore(object):
...
+            else self.data.dtype
+        )
+        if isinstance(nodata, float):
+            if dtype != "float32":
+                dtype = "float64"
+
+        if self.scatter is None:
+            band_count = self.data.gw.nbands
+        else:
+            if self.scatter == 'band':
```
Comment (jgrss) [inline src/geowombat/backends/rasterio_.py:219]: File scattering is new.

File: tests/test_stac.py [hunk 5]
```diff
@@ -179,65 +229,78 @@ def test_download_blue_sentinel_s2_l2a(self):
...
+
+            out_path_client = Path(tmp_path) / 'test_client.tif'
+            with LocalCluster(
+                processes=True,
+                n_workers=2,
+                threads_per_worker=1,
+                memory_limit="1GB",
+            ) as cluster:
+                with Client(cluster) as client:
+                    time_mean.gw.save(
```
Comment (jgrss) [inline tests/test_stac.py:263]: Here is the test using a client.

File: src/geowombat/core/geoxarray.py [hunk 6]
```diff
@@ -728,34 +728,39 @@ def to_netcdf(
...
+            scatter (Optional[str]): Scatter 'band' or 'time' to separate file. Default is None.
             client (Optional[Client object]): A ``dask.distributed.Client`` client object to persist data.
                 Default is None.
             compute (Optinoal[bool]): Whether to compute and write to ``filename``. Otherwise, return
                 the ``dask`` task graph. If ``True``, compute and write to ``filename``. If ``False``,
                 return the ``dask`` task graph. Default is ``True``.
             tags (Optional[dict]): Metadata tags to write to file. Default is None.
             compress (Optional[str]): The file compression type. Default is 'none', or no compression.
+
+                .. note::
```
Comment (jgrss) [inline src/geowombat/core/geoxarray.py:759]: @mmann1123 👀

File: tests/test_write.py [hunk 7]
```diff
@@ -142,6 +143,76 @@ def test_save(self):
...
+                    self.assertTrue(hasattr(tmp_src, "TEST_METADATA"))
+                    self.assertEqual(tmp_src.TEST_METADATA, "TEST_VALUE")
+
+            with rio.open(out_path) as rio_src:
+                self.assertTrue(rio_src.nodata == NODATA)
+
+    def test_client_save(self):
+
+        with LocalCluster(
+            processes=False,
```
Comment (jgrss) [inline tests/test_write.py:181]: Necessary to write to compressed file.

File: tests/test_write.py [hunk 8]
```diff
@@ -142,6 +143,76 @@ def test_save(self):
...
+            with Client(cluster) as client:
+                with tempfile.TemporaryDirectory() as tmp:
+                    out_path = Path(tmp) / "test.tif"
+                    with gw.open(l8_224078_20200518) as src:
+                        data = src.gw.set_nodata(0, NODATA, dtype="uint16")
+                        data.gw.save(
+                            filename=out_path,
+                            overwrite=True,
+                            tags={"TEST_METADATA": "TEST_VALUE"},
+                            compress="lzw",
```
Comment (jgrss) [inline tests/test_write.py:195]: Compression should be okay.

**Comment by jgrss** [comment]
I merged this into #319.

**Pull request closed without merging**